### PR TITLE
Get GPIO pin ID from controller 

### PR DESCRIPTION
**Pull request by pallaud**
Instead of having to use hardcoded pin IDs with the gpio_linux package, we can compute the pin ID from the gpiochip directories (controllers)

### CODE REVIEW

File: pkg/gpio/gpio_linux.go [hunk 1]
```diff
@@ -39,10 +40,49 @@ func (v Value) String() string {
 	return "1"
 }
 
+// GetPinID is used to get the ID for a specific GPIO. This is done by
```
Comment (hugelgupf) [inline pkg/gpio/gpio_linux.go]: the first sentence should be an active voice summary of what this does, e.g.

```go
// GetPinID computes the sysfs pin number for a specific port on a specific GPIO chip.
```

File: pkg/gpio/gpio_linux.go [hunk 2]
```diff
@@ -39,10 +40,49 @@ func (v Value) String() string {
...
+	}
+
+	return 0, fmt.Errorf("could not find controller %s", controller)
+}
+
 // SetOutputValue configures the gpio as an output pin with the given value.
 func SetOutputValue(pin int, val Value) error {
 	dir := val.Dir()
-	path := filepath.Join(gpioPath, "gpio", strconv.Itoa(pin), "direction")
+	path := filepath.Join(gpioPath, fmt.Sprintf("gpio%d", pin), "direction")
```
Comment (hugelgupf) [inline pkg/gpio/gpio_linux.go]: did this not already go in?

Comment (pallaud) [reply pkg/gpio/gpio_linux.go]: It should have, I can see it in master, I'm not sure why that commit is being added here...

File: pkg/gpio/gpio_linux.go (hunk 1 above)
Comment (pallaud) [reply pkg/gpio/gpio_linux.go]: Done

File: pkg/gpio/gpio_linux.go [hunk 3]
```diff
@@ -39,6 +40,45 @@ func (v Value) String() string {
 	return "1"
 }
 
+// GetPinID computes the sysfs pin ID for a specific port on a specific GPIO
+// controller chip. It gets the base offset of the chip, and adds the specific
+// pin number.
```
Comment (hugelgupf) [inline pkg/gpio/gpio_linux.go]: add: The controller arg is matched to a gpiochip's label in sysfs.

**Comment by hugelgupf** [comment]
Filed #1327 for the test failure

File: pkg/gpio/gpio_linux.go (hunk 3 above)
Comment (pallaud) [reply pkg/gpio/gpio_linux.go]: done

**Pull request merged**